DataFormats/DTDigi/interface/DTuROSControlData.h

```diff
@@ -0,0 +1,133 @@
+//-------------------------------------------------
+//
+/**  \class DTuROSControlData
+ *
+ *   DT uROS Control Data
+ *
+ *
+ *
+ *
+ *   J. Troconiz  - UAM Madrid
+ */
+//
+//--------------------------------------------------
+#ifndef DTDigi_DTuROSControlData_h
+#define DTDigi_DTuROSControlData_h
+
+#include <vector>
+
+
+class DTuROSROSData {
+
+public:
+
+  /// Constructor
+  DTuROSROSData() {slot_ = -1;}
+
+  /// Destructor
+  ~DTuROSROSData(){};
+
+  void setslot(int slot) {slot_ = slot;}
+
+  void setheader1(long dword) {header1_ = dword;}
+
+  void setheader2(long dword) {header2_ = dword;}
+
+  void settrailer(long dword) {trailer_ = dword;}
+
+  void setokword1(int okword) {okword1_ = okword;}
+
+  void setokword2(int okword) {okword2_ = okword;}
+
+  void seterror(int error) {error_.push_back(error);}
+
+  int getslot() const {return slot_;}
+
+  long getheader1() const {return header1_;}
+
+  long getheader2() const {return header2_;}
+
+  long gettrailer() const {return trailer_;}
+
+  int getokword1() const {return okword1_;}
+
+  int getokword2() const {return okword2_;}
+
+  int getokflag(int i) const {if (i < 60) return ((okword1_ >> i)&0x1); return ((okword2_ >> (i-60))&0x1);}
+
+  std::vector<int> geterrors() const {return error_;}
+
+  int geterror(int i) const {return error_.at(i);}
+
+  int geterrorROBID(int i) const {return (error_.at(i) >> 21)&0x7F;}
+
+  int geterrorTDCID(int i) const {return (error_.at(i) >> 19)&0x3;}
+
+  int geterrorFlag(int i) const {return (error_.at(i))&0x7FFF;}
+
+private:
+
+  long header1_, header2_, trailer_;
+
+  int slot_, okword1_, okword2_;
+
+  std::vector<int> error_;
+
+};
+
+
+class DTuROSFEDData {
+
+public:
+
+  /// Constructor
+  DTuROSFEDData() {for (int i=0; i<12; i++) rsize_[i] = 0;} 
```

Comment:
Is this "12" defined anywhere, or can it be extracted from some data format?

Reply:
done.